File: src/main/java/org/pulcifer/climate/repo/CityRepository.java
package org.pulcifer.climate.repo;

import org.pulcifer.climate.model.City;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;
import java.util.Optional;

public interface CityRepository extends MongoRepository<City, String> {
    Optional<City> findByCityId(Integer cityId);
    Optional<List<City>> findByCityIdNot(Integer cityId);
    Optional<List<City>> findByCityIdIn(List<Integer> cityIds);
    Optional<City> findByCityName(String cityName);
    Optional<List<City>> findByCityNameContaining(String cityName);
}

File: src/main/kotlin/org/pulcifer/climate/repo/CityRepository.kt
package org.pulcifer.climate.repo

import org.pulcifer.climate.model.City
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.*

interface CityRepository : MongoRepository<City?, String?> {
    fun findByCityId(cityId: Int?): Optional<City?>
    fun findByCityIdNot(cityId: Int?): Optional<List<City?>>
    fun findByCityIdIn(cityIds: List<Int?>?): Optional<List<City?>>
    fun findByCityName(cityName: String?): Optional<City?>
    fun findByCityNameContaining(cityName: String?): Optional<List<City?>?>
}